### ISSUE TITLE
Can't sign GALXE transactions on Ledger on Optimism

### ISSUE THREAD

**Issue by Naxsun** [tallycash/extension#2299]
### Discord Discussion Link

https://discord.com/channels/808358975287722045/888500174685614090/1023491062926417950

### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

Sign a signature message through the Ledger

### What did not work?

To reproduce:
- Connect to GALXE with a ledger address 
- Set the network in the Extension to Optimism (I believe Polygon has the same behavior)
- Go to Galxe ID, profile
- Select 'Edit'
- It will require you to sign a message. The initial screen comes up in Tally Ho, the Ledger sign screen in Tally Ho briefly flashes, but then jumps back
- There is no action on the Ledger 
- 
<img width="1655" alt="Screen Shot 2022-09-26 at 14 02 26" src="https://user-images.githubusercontent.com/71384922/192271482-c136437e-a85c-43b4-a6a8-c30354904536.png">

### Version

v0.16.2

**Comment by beemeeupnow** [comment]
still present in 0.46.1
 
When the dapp selected network is not Ethereum and the SIWE request comes, clicking on the Sign button generates an error when sending to the Ledger device.
  
relevant stack:
```
Async thunk failed Error: Unsupported network for Ledger signing
    at LedgerService.signMessage (index.ts:550:13)
    at SigningService.signData (index.ts:311:49)
    at main.ts:936:54
    at index.js:306:13
    at Array.map (<anonymous>)
    at Emittery.emit (index.js:304:23)
    at async signing.ts:86:40
    at async utils.ts:139:9
logEvent @ logger.ts:208
```

**Comment by Shadowfiend** [comment]
What a coincidence, I'm looking at this right now!

Tl;dr here is that we decide whether or not a Ledger can sign messages based on a hardcoded set of networks that is just Ethereum mainnet and Polygon. A better way to do this is to use derivation paths, and in particular making sure that the derivation path used by the current network matches the one used by the app on the Ledger. However, derivation paths are internally optional in the code, and the only network that has a derivation path set is actually RSK.

I'm working through some of this at the moment as part of implementing control compatibility ([RFB 5](https://github.com/tahowallet/extension/blob/main/rfb/rfb-5-control-compatile-addresses.md)), but my bandwidth is pretty constrained… Hoping to get a small chunk of this emitted as a bug fix for this particular issue in the next couple of weeks though, fingers crossed!

**Comment by beemeeupnow** [comment]
Would it be possible to implement some workaround until a complete fix can be delivered?

This issue can be a big pain if the dapp doesn't have a network selector since the wallet 'remembers' what network was used on the dapp connection.

Even if the user:
* disconnects with the lightning bolt button in the wallet
* clears browser cache
* restarts browser
* changes network in wallet to Ethereum
* connects to the dapp again

the wallet still uses the wrong network since it pulls it from storage. (_Major_ UX showstopper here)
